sycl/doc/extensions/experimental/sycl_ext_oneapi_memory_export.asciidoc

```diff
@@ -0,0 +1,338 @@
+
+```c++
+namespace sycl::ext::oneapi::experimental {
+
+// External memory resource handle types.
+enum class export_external_mem_handle_type {
+  opaque_fd = 0,
+  win32_nt = 1,
+};
+
+template <export_external_mem_handle_type ExportMemHandleType>
+struct exported_mem;
+
+template <>
+struct exported_mem<export_external_mem_handle_type::opaque_fd> {
+  using type = int;
+};
+
+template <>
+struct exported_mem<export_external_mem_handle_type::win32_nt> {
+  using type = void *;
+};
+
+template <export_external_mem_handle_type ExportMemHandleType>
+using exported_mem_t = typename exported_mem<ExportMemHandleType>::type;
+
+}
+```
+
+The `export_external_mem_handle_type` enum class defines the types of external
+memory resource handles that can be exported by this extension. The `opaque_fd`
+and `win32_nt` values are used during allocation of exportable memory to
+indicate the type of handle that will later be returned by the
+`export_device_mem_handle` function.
+
+The `opaque_fd` handle type corresponds to a POSIX file descriptor, which is
+represented by an `int`.
+
+The `win32_nt` handle type corresponds to a Windows NT handle, which is
+represented by a `void *`.
+
+The `exported_mem` template struct is specialized for each of the
+`export_external_mem_handle_type` values. Each specialization defines a `type`
+alias that represents the type of the exported memory handle for that specific
+handle.
+
+The `exported_mem_t` type alias allows the user to easily obtain the type of
+the exported memory handle based on the `export_external_mem_handle_type`
+template parameter.
+
+=== API of the extension
+
+```c++
+
+namespace sycl::ext::oneapi::experimental {
+
+void *alloc_exportable_device_mem(
+    size_t alignment, size_t size,
+    export_external_mem_handle_type exportMemHandleType,
+    const sycl::device &syclDevice, const sycl::context &syclContext,
+    const property_list& propList = {});
+
+void *alloc_exportable_device_mem(
+    size_t alignment, size_t size,
+    export_external_mem_handle_type exportMemHandleType,
+    const sycl::queue &syclQueue,
+    const property_list& propList = {});
+}
+```
+
+The `alloc_exportable_device_mem` function allocates memory on the device marked
+as having the ability to later export that memory to an external memory resource
+handle.
+
+Memory allocated through this function must only be freed using
+`free_exportable_mem`. Using `sycl::free` to deallocate memory allocated with
+this function results in undefined behavior.
+
+With the exception of the `sycl::free` function from the core SYCL
+specification, pointers to memory allocated through this function may be passed
+to any core SYCL specification API accepting device USM memory pointers.
+
+Memory allocated through this function is only available on device.
+
+Memory allocated through this function has a linear memory layout on the device 
+(which is the same as memory allocated by other USM allocation functions like 
+`sycl::malloc_device`).
```

Comment:
Just highlighting this to point out that you wouldn't have to specify all of this behavior if we were modifying `malloc_device` instead of introducing a new allocation function.

Reply:
See the comment and linked discussions below on this issue:

https://github.com/intel/llvm/pull/19018#discussion_r2182052526